lalrpop-util/src/lib.rs

```diff
@@ -108,14 +118,32 @@ impl<L, T, E> ParseError<L, T, E> {
         }
     }
 
+    /// Transform a ParseError by applying a function to the location field
+    ///
+    /// This could be useful for example to obtain a `ParseError` where all
+    /// fields implement `Display` or `Debug`.  `ParseErrors` where all fields
+    /// implement `Display` also implement `Display` and `ParseErrors` where all
+    /// fields implement both `Display` and `Debug` implement `Error`
     pub fn map_location<LL>(self, op: impl FnMut(L) -> LL) -> ParseError<LL, T, E> {
```

Comment:
One thing I find interesting is that location takes a FnMut compared to the other two functions(Because it is applied to both the start and ending locations).

Reply:
Yeah, interesting.  I added a comment on the maptok declaration above, as well as a note here, since that's not immediate obvious when looking at the code.  I'm a little torn on what level of detail the user-facing docs need.  On the one hand I can imagine user confusion over the signature differences, but on the other hand, the reason it needs to be `FnMut` instead of `FnOnce` is arguable more of an implementation detail.

What do you think of the way it is now?

Reply:
Simpler but more clear in my opinion.